app/helpers/payouts_helper.rb

```diff
@@ -122,6 +236,22 @@ def old_payout_period_data(user:, payment:)
       payout_period_data[:direct_fees_cents] = payout_period_data[:direct_fees_cents].to_i + payment.gumroad_fee_cents
     end
 
+    if balance_ids.present? && payment.failed?
+      later_completed_list = user.payments
+        .completed
+        .joins(:balances)
+        .where("payments.created_at > ?", payment.created_at)
+        .where(balances: { id: balance_ids })
+        .order("payments.created_at ASC")
+        .distinct
+
+      if later_completed_list.exists?
+        payout_period_data[:included_in_payout_on_items] = later_completed_list.map do |lp|
```

Comment:
Think balance from a failed payout will be included in only 1 future payout, there shouldn't be more than 1 such payouts.